### PR TITLE
Force usage of VS2019 and revert VS 2022 support

### PR DESCRIPTION
**Pull request by hoyosjs**
CMake doesn't yet support VS2022 and several team members have complained this change broke building the repo on their machine if VS2022 was installed SxS. 

cc: @tommcdon

### CODE REVIEW

**Comment by BrennanConroy** [comment]
And now you're breaking people that have 2022 only. Not my decision to make, just letting you know.

**Comment by stephentoub** [comment]
> CMake doesn't yet support VS2022 and several team members have complained this change broke building the repo on their machine if VS2022 was installed SxS.

Is this the cause of the debugging failures when trying to debug dotnet/runtime tests in VS2019? Note I get such failures and I don't have VS2022 installed.

**Comment by hoyosjs** [comment]
The native portion of the runtime relies and needs VS 2019 until CMake supports 2022. I tried scouting VS2022, but support is not there and there's at least 3 team members that have reported being unable to build because of this. If anything,  the VS requirement between libraries and runtime could be separated, but given libraries depends on coreclr/mono I am not sure we can do this. 

> Is this the cause of the debugging failures when trying to debug dotnet/runtime tests in VS2019? Note I get such failures and I don't have VS2022 installed.

No. The debugger is actually currently broken. Trying to build a debug build right now to fix that.

**Comment by jkoritzinsky** [comment]
@hoyosjs is the default Ninja experience broken, or is just the MSBuild/VS generator experience not working?

**Comment by hoyosjs** [comment]
For me usual works just fine. I know two of the issues that got reported were people that use the solution. The other person was using the vainilla build, but I don't know what commit they are on nor if they use the VS provided ninja.

**Comment by jkotas** [comment]
If you run the build from the VS command prompt for your VS version of choice, it is expected to pick up that VS version and not try to autodetect (it happens here: https://github.com/dotnet/runtime/pull/53554/files#diff-0b2616a9a39ad5f73a5f4bd87b0427f8f6d258114ec1971d6e24511014983286R24). Could you please make sure that this still works for VS2022? It is important to allow people to dogfood VS2022.

File: eng/native/init-vs-env.cmd [hunk 1]
```diff
@@ -50,14 +50,9 @@ if "%VisualStudioVersion%"=="16.0" (
     set __PlatformToolset=v142
     goto :SetVCEnvironment
 )
-if "%VisualStudioVersion%"=="17.0" (
```
Comment (jkotas) [inline eng/native/init-vs-env.cmd:53]: I think this should stay to enable VS2022 dogfooding via running from VS2022 command prompt.

**Comment by jkotas** [comment]
Also, it may be convenient to use VS2022 as fallback. When both VS2019 and VS2022 are installed, make the auto-detection preffer VS2019.

**Comment by hoyosjs** [comment]
> Could you please make sure that this still works for VS2022? It is important to allow people to dogfood VS2022.

It wasn't working according to Tom who hit the issue. I don't understand how as defining VisualStudioVersion should skip the detection logic and VS 2022 sets it. I'll dig into this more.

**Comment by hoyosjs** [comment]
I'll close this as the more I think about, the more this just feels like a wrong reactionary fix.

**Pull request closed without merging**